Support qmatmul with different dims tensors

mcremon-meta opened the pull request:
Summary:
MobileBERT exposes an issue in our kernel, where tensors have compatible (for PyTorch) but different batch dimensions.

This diff changes the meta kernel to support that (the kernel can already do it).

Differential Revision: D60314979

dulinriley commented on backends/cadence/aot/ops_registrations.py:
```diff
@@ -186,28 +187,29 @@ def quantized_matmul_meta(
+        assert X_size[-1] == Y_size[-2], "matrices cannot be multiplied"
+        mat_size = [X_size[-2], Y_size[-1]]
+
+    # Combine the larger batch dimensions with the matmul output size
+    out_size = (
+        X_batch_dims + mat_size
+        if len(X_batch_dims) > len(Y_batch_dims)
+        else Y_batch_dims + mat_size
+    )
```
Why combine the larger of the two batch dims? Perhaps it should just default to the first argument's instead?
This would seem like unpredictable behavior to someone using this function.

mcremon-meta committed:
Support qmatmul with different dims tensors (#4438)

Summary:
Pull Request resolved: https://github.com/pytorch/executorch/pull/4438

MobileBERT exposes an issue in our kernel, where tensors have compatible (for PyTorch) but different batch dimensions.

This diff changes the meta kernel to support that (the kernel can already do it).

Reviewed By: dulinriley

Differential Revision: D60314979
